### PR TITLE
api: remove the version parameter of getSupportInfo

### PR DESCRIPTION
**Pull request by thorn0**
closes https://github.com/prettier/prettier/issues/7615

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/main/support.js [hunk 1]
```diff
@@ -9,30 +9,26 @@ const arrayify = require("../utils/arrayify");
...
+    showUnreleased = false,
+    showDeprecated = false,
+    showInternal = false
   }
-
-  const { plugins } = opts;
+) {
+  // pre-release version is smaller than the normal version in semver,
+  // we need to treat it as the normal one so as to test new features.
+  const version = currentVersion.split("-", 1)[0];
```
Comment (fisker) [inline src/main/support.js]: I never know split takes a second argument.

Comment (thorn0) [reply src/main/support.js]: It's not super useful anyway.

**Comment by fisker** [comment]
We may want switch those support infos to JSON scheme some day.

**Comment by thorn0** [comment]
@fisker Would we need that `version` parameter for that?

**Comment by fisker** [comment]
I agree that we don't need it. I mean after removing this, support info is just some property validate info, we can use ajv to validate.

File: src/main/support.js [hunk 2]
```diff
@@ -9,30 +9,26 @@ const arrayify = require("../utils/arrayify");
...
-    showInternal: false,
-    ...opts
-  };
-
-  if (!version) {
-    // pre-release version is smaller than the normal version in semver,
-    // we need to treat it as the normal one so as to test new features.
-    version = currentVersion.split("-", 1)[0];
+function getSupportInfo(
+  _,
```
Comment (fisker) [inline src/main/support.js]: Can we remove this parameter, and wrap in the export?

Comment (thorn0) [reply src/main/support.js]: That would break type inference and overall I think the code is cleaner (less confusing) the way it is now.

Comment (thorn0) [reply src/main/support.js]: Nevertheless, I did something with this.

**Comment by thorn0** [comment]
Merging this to move forward with other changes.

**Pull request merged**